### PR TITLE
SBERDOMA-63 createservice tool

### PR DESCRIPTION
**Pull request by MrFoxPro**

### CODE REVIEW

**Review by dkoviazin — COMMENTED** [review]
Please, include a patch to packages/@core.keystone/schema.js for class GQLCustomSchema
with queries from @Dimitreee to this PR

File: packages/@core.codegen/index.js [hunk 1]
```diff
@@ -183,7 +190,7 @@ async function generate (templateDirectory, targetDirectory, ctx) {
     await renderTemplates(templateDirectory, tmpDirectory, ctx)
     await renaming(templateDirectory, tmpDirectory, ctx)
 
-    const isPatchFile = (name) => !name.endsWith('.patch') && !name.endsWith('.default')
+    const isPatchFile = (name) => !name.endsWith('.patch') && !name.endsWith('.default') && !name.includes('.example.')
```
Comment (MrFoxPro) [inline packages/@core.codegen/index.js:193]: In future I have idea to add .example file for every template

**Comment by MrFoxPro** [comment]
As I understood, .default tempaltes both for schema and service should be equal, because current implementation of codegen tool can't merge them. So every scheme without service will contain unused imports and otherwise.

**Review by dkoviazin — APPROVED** [review]
Ok

**Comment by MrFoxPro** [comment]
Usage example:
First example (Schema then service):
`yarn workspace @app/condo createschema example.Example "name:Text; email:Text; password?:Password; isAdmin:Checkbox; manager?"`
`yarn workspace @app/condo createservice example.manageExampleService`

Second example (only service):
`yarn workspace @app/condo createservice property.CheckAddressExistsService`

**Review by Dimitreee — APPROVED** [review]

File: packages/@core.codegen/templates/service00/domains/{{domain}}/access/{{name}}.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Generated by `{{ command }}`
+ */
+
+async function can{{ name.replace('Service', '') }} ({ authentication: { item: user } }) {
+    if (!user) return false
```
Comment (pahaz) [inline packages/@core.codegen/templates/service00/domains/{{domain}}/access/{{name}}.js:6]:
```suggestion
    if (!user) return throwAuthenticationError()
```

File: packages/@core.codegen/templates/service00/domains/{{domain}}/access/{{name}}.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Generated by `{{ command }}`
+ */
+
```
Comment (pahaz) [inline packages/@core.codegen/templates/service00/domains/{{domain}}/access/{{name}}.js:4]:
```suggestion
const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')

```

File: packages/@core.codegen/index.js [hunk 4]
```diff
@@ -131,7 +134,8 @@ async function renaming (templateDirectory, targetDirectory, ctx) {
     const files = await readdir(targetDirectory, { withFileTypes: true })
     for (const file of files) {
         const filename = file.name
-        const renderedName = renderToString(filename, filename, ctx)
+        let renderedName = renderToString(filename, filename, ctx)
+        if (ctx.opts.type) renderedName = renderedName.replace('.' + ctx.opts.type, '')
```
Comment (pahaz) [inline packages/@core.codegen/index.js:138]: It’s redundant. It’s better to just use if statement in template language

File: packages/@core.codegen/templates/service00/domains/{{domain}}/schema/{{name}}.js.queries [hunk 5]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Generated by `{{ command }}`
+ */
+
+const { GQLCustomSchema } = require('@core/keystone/schema')
+const access = require('@{{app}}/domains/{{ domain }}/access/{{name}}')
+
+
+const {{ name }} = new GQLCustomSchema('{{ name }}', {
+    types: [
```
Comment (pahaz) [inline packages/@core.codegen/templates/service00/domains/{{domain}}/schema/{{name}}.js.queries:10]: It’s better to just use if statement for this case.

**Pull request merged**